stock_picking_compute_delivery_date/model/picking.py

```diff
@@ -113,12 +124,17 @@ def compute_mto_delivery_dates(self, cr, uid, product, context=None):
                 }, context=context)
                 message = _("Scheduled date update to %s from %s") \
                     % (new_date_str, move_in.picking_id.name)
-                # Post message on stock picking
-                self.message_post(cr, uid, move_out.picking_id.id,
-                                  body=message, context=context)
+                # Add message on stock picking:
+                # - if it does not exist, add the current one
+                # - if it does, overwrite it, and keep the "last" one.
+                if 'messages' not in context:
+                    context['messages'] = {}
```

Comment:
Please don't assign context it's dangerous, your changes on a mutable will be done for all calling methods.

Plus this won't work when it will be ported to v8.0 api as context will be a frozendict.